internal/stage_magnet_reserved.go

```diff
@@ -41,10 +41,18 @@ func testMagnetReserved(stageHarness *test_case_harness.TestCaseHarness) error {
     return nil
 }
 
-func handleReservedBytes(conn net.Conn, params PeerConnectionParams) {
+func handleReservedBytes(conn net.Conn, p PeerConnectionParams) {
     defer conn.Close()
 
-    if err := receiveAndSendHandshake(conn, params); err != nil {
+    if err := receiveAndSendHandshake(conn, p); err != nil {
+        return
+    }
+
+    if err := sendBitfieldMessage(conn, p.bitfield, p.logger); err != nil {
```

Comment:
adding these because clients will wait for these messages as they build the next steps of handshake